src_features/provideDomainName/cmd_provide_domain_name.c

```diff
@@ -0,0 +1,567 @@
+#include "domain_name.h"
+#include "challenge.h"
+#include "mem.h"
+#include "hash_bytes.h"
+
+static const uint8_t DOMAIN_NAME_PUB_KEY[] = {
+#ifdef HAVE_DOMAIN_NAME_TEST_KEY
+    0x04, 0xb9, 0x1f, 0xbe, 0xc1, 0x73, 0xe3, 0xba, 0x4a, 0x71, 0x4e, 0x01, 0x4e, 0xbc,
+    0x82, 0x7b, 0x6f, 0x89, 0x9a, 0x9f, 0xa7, 0xf4, 0xac, 0x76, 0x9c, 0xde, 0x28, 0x43,
+    0x17, 0xa0, 0x0f, 0x4f, 0x65, 0x0f, 0x09, 0xf0, 0x9a, 0xa4, 0xff, 0x5a, 0x31, 0x76,
+    0x02, 0x55, 0xfe, 0x5d, 0xfc, 0x81, 0x13, 0x29, 0xb3, 0xb5, 0x0b, 0xe9, 0x91, 0x94,
+    0xfc, 0xa1, 0x16, 0x19, 0xe6, 0x5f, 0x2e, 0xdf, 0xea
+#else
+    0x04, 0x6a, 0x94, 0xe7, 0xa4, 0x2c, 0xd0, 0xc3, 0x3f, 0xdf, 0x44, 0x0c, 0x8e, 0x2a,
+    0xb2, 0x54, 0x2c, 0xef, 0xbe, 0x5d, 0xb7, 0xaa, 0x0b, 0x93, 0xa9, 0xfc, 0x81, 0x4b,
+    0x9a, 0xcf, 0xa7, 0x5e, 0xb4, 0xe5, 0x3d, 0x6f, 0x00, 0x25, 0x94, 0xbd, 0xb6, 0x05,
+    0xd9, 0xb5, 0xbd, 0xa9, 0xfa, 0x4b, 0x4b, 0xf3, 0xa5, 0x49, 0x6f, 0xd3, 0x16, 0x4b,
+    0xae, 0xf5, 0xaf, 0xcf, 0x90, 0xe8, 0x40, 0x88, 0x71
+#endif
+};
+
+#define P1_FIRST_CHUNK     0x01
+#define P1_FOLLOWING_CHUNK 0x00
+
+#define ALGO_SECP256K1 1
+
+#define SLIP_44_ETHEREUM 60
+
+typedef enum { TLV_TAG, TLV_LENGTH, TLV_VALUE } e_tlv_step;
+
+typedef enum {
+    STRUCTURE_TYPE = 0x01,
+    STRUCTURE_VERSION = 0x02,
+    CHALLENGE = 0x12,
+    SIGNER_KEY_ID = 0x13,
+    SIGNER_ALGO = 0x14,
+    SIGNATURE = 0x15,
+    DOMAIN_NAME = 0x20,
+    COIN_TYPE = 0x21,
+    ADDRESS = 0x22
+} e_tlv_tag;
+
+typedef enum { KEY_ID_TEST = 0x00, KEY_ID_PROD = 0x03 } e_key_id;
+
+typedef struct {
+    uint8_t *ptr;
+    uint16_t size;
+    uint16_t expected_size;
+} s_tlv_payload;
+
+typedef struct {
+    e_tlv_tag tag;
+    uint8_t length;
+    const uint8_t *value;
+} s_tlv_data;
+
+typedef struct {
+    bool valid;
+    char *name;
+    uint8_t addr[ADDRESS_LENGTH];
+} s_domain_name_info;
+
+typedef struct {
+    e_key_id key_id;
+    uint8_t input_sig_size;
+    const uint8_t *input_sig;
+    cx_sha256_t hash_ctx;
+} s_sig_ctx;
+
+typedef bool (*t_tlv_handler)(const s_tlv_data *data,
+                              s_domain_name_info *domain_name_info,
+                              s_sig_ctx *sig_ctx);
+
+typedef struct {
+    uint8_t tag;
+    t_tlv_handler func;
+    uint8_t found;
+} s_tlv_handler;
+
+static s_tlv_payload g_tlv_payload = {0};
+static s_domain_name_info g_domain_name_info;
+char domain_name[DOMAIN_NAME_MAX_LENGTH + 1];
+
+/**
+ * Send a response APDU
+ *
+ * @param[in] success whether it should use \ref APDU_RESPONSE_OK
+ * @param[in] off payload offset (0 if no data other than status word)
+ */
+static void response_to_domain_name(bool success, uint8_t off) {
+    uint16_t sw;
+
+    if (success) {
+        sw = APDU_RESPONSE_OK;
+    } else {
+        sw = apdu_response_code;
+    }
+    U2BE_ENCODE(G_io_apdu_buffer, off, sw);
+
+    io_exchange(CHANNEL_APDU | IO_RETURN_AFTER_TX, off + 2);
+}
+
+/**
+ * Verify the domain name in memory with the given address and chain ID
```

Comment:
`verify` implies an error / succes check, here it's only an informative check
I suggest something like `is_domain_name_known()`